src/main/java/io/jenkins/plugins/sshbuildagents/ssh/mina/SSHApacheMinaLauncher.java

```diff
@@ -0,0 +1,1125 @@
+        }
+        StandardUsernameCredentials credentials = getCredentials();
+        if (credentials == null) {
+            throw new AbortException("Cannot find SSH User credentials with id: " + credentialsId);
+        }
+        // TODO implement verifiers.
+        String[] preferredKeyAlgorithms =
+                getSshHostKeyVerificationStrategyDefaulted().getPreferredKeyAlgorithms(computer);
+        if (preferredKeyAlgorithms != null && preferredKeyAlgorithms.length > 0) { // JENKINS-44832
+            connection.setServerHostKeyAlgorithms(preferredKeyAlgorithms);
+        } else {
+            listener.getLogger().println("Warning: no key algorithms provided; JENKINS-42959 disabled");
+        }
+        PrintStream logger = listener.getLogger();
+        logger.println(Messages.SSHLauncher_OpeningSSHConnection(getTimestamp(), host + ":" + port));
+        connection.setTCPNoDelay(getTcpNoDelay());
+        connection.setServerHostKeyVerifier(new ServerHostKeyVerifierImpl(computer, listener));
+        connection.setTimeout((int) getLaunchTimeoutMillis());
+        connection.setCredentials(credentials);
+        connection.setRetries(getMaxNumRetries());
+        connection.setRetryWaitTime(getRetryWaitTime());
+        connection.setWorkingDirectory(workingDirectory);
+        connection.setStdErr(new NoCloseOutputStream(listener.getLogger()));
+        connection.setStdOut(new NoCloseOutputStream(listener.getLogger()));
+        connection.connect();
+    }
+
+    /**
+     * Validates the Agent configuration.
+     *
+     * @throws InterruptedException
+     */
+    private void checkConfig() throws InterruptedException {
+        // JENKINS-58340 some plugins does not implement Descriptor
+        Descriptor descriptorOrg = Jenkins.get().getDescriptor(this.getClass());
+        if (!(descriptorOrg instanceof DescriptorImpl)) {
+            return;
+        }
+
+        DescriptorImpl descriptor = (DescriptorImpl) descriptorOrg;
+        String message = "Validate configuration:\n";
+        boolean isValid = true;
+
+        String port = String.valueOf(this.port);
+        FormValidation validatePort = descriptor.doCheckPort(port);
+        FormValidation validateHost = descriptor.doCheckHost(this.host);
+        FormValidation validateCredentials =
+                descriptor.doCheckCredentialsId(Jenkins.get(), Jenkins.get(), this.host, port, this.credentialsId);
+
+        if (validatePort.kind == FormValidation.Kind.ERROR) {
+            isValid = false;
+            message += validatePort.getMessage() + "\n";
+        }
+        if (validateHost.kind == FormValidation.Kind.ERROR) {
+            isValid = false;
+            message += validateHost.getMessage() + "\n";
+        }
+        if (validateCredentials.kind == FormValidation.Kind.ERROR) {
+            isValid = false;
+            message += validateCredentials.getMessage() + "\n";
+        }
+
+        if (!isValid) {
```

Comment:
Just a nit, but if the configuration is valid, won't the output just be:
`Validate configuration:`
with nothing else?